test/prototype_transforms_kernel_infos.py

```diff
@@ -2100,3 +2118,39 @@ def sample_inputs_convert_dtype_video():
         ),
     ]
 )
+
+
+def sample_inputs_uniform_temporal_subsample_video():
+    for video_loader in make_video_loaders(sizes=["random"], num_frames=[4]):
+        for temporal_dim in [-4, len(video_loader.shape) - 4]:
+            yield ArgsKwargs(video_loader, num_samples=2, temporal_dim=temporal_dim)
+
+
+def reference_uniform_temporal_subsample_video(video, num_samples, temporal_dim=-4):
+    from pytorchvideo.transforms.functional import uniform_temporal_subsample
+
+    return uniform_temporal_subsample(video, num_samples=num_samples, temporal_dim=temporal_dim)
+
+
+def reference_inputs_uniform_temporal_subsample_video():
+    for video_loader in make_video_loaders(sizes=["random"], color_spaces=[features.ColorSpace.RGB], num_frames=[10]):
+        for num_frames in range(1, 11):
+            yield ArgsKwargs(video_loader, num_frames)
```

Comment:
The reference test as a separate set of inputs that will be tested

https://github.com/pytorch/vision/blob/9c112935abe400222cca8f9fbc2d8386e0f25e80/test/prototype_transforms_kernel_infos.py#L50-L52

That has two reasons:

1. Sometimes the reference doesn't support all the inputs our kernel supports. For example, using a PIL kernel as reference requires the input to be unbatched and thus you will often see `extra_dims=[()]` in the reference inputs for kernels that have a PIL reference.
2. There is only one reference test and thus all the considerations of keeping the number of inputs low that we made for the sample inputs are off the table here. This can be as comprehensive as we need it for us to be sure our kernel is doing the right thing.

Of course the kernel under test here is pretty simple and thus the reference inputs don't look that much different from the sample inputs. I again opted to use a fixed spatial size and number of frames. This time I also used only one color space since and one temporal dim (the default one) since any errors regarding that will already be caught by the other tests. In contrast, I here varied the number of frames that will be subsampled very fine grained to make sure our rounding is doing exactly the same as the reference.